### PR TITLE
Use latest client package in Dockerfiles and automatically build+publish client container

### PR DESCRIPTION
**Pull request by bedroge**
Tested on x86_64. Updated containers do still need to be pushed them to Docker Hub. (we really need to automate this...)

### CODE REVIEW

**Comment by boegel** [comment]
@bedroge Isn't this a good opportunity to look into automating the build of containers, and use the GitHub container registry instead?

**Comment by bedroge** [comment]
> @bedroge Isn't this a good opportunity to look into automating the build of containers, and use the GitHub container registry instead?

I've added a workflow that builds the client container image and pushes it to the GitHub container registry. It also uses multi-arch images now (one image name for all architectures), and a multi-stage build so that all build requirements for creating the RPMs do not have to be included in the client container. The RPMs for `ppc64le` have to be built from source, and that takes a while (~2h).

**Comment by bedroge** [comment]
Does it make sense to still keep the `test-containers.yml` workflow (which still runs on every push/pr, only for `x86_64`), or should I remove that one?

**Comment by bedroge** [comment]
For an example of how this multiarch container looks like, see:
https://github.com/users/bedroge/packages/container/package/client-pilot

**Comment by bedroge** [comment]
Successfully tested on `x86_64` and `ppc64le` using:
```
mkdir -p /tmp/$USER/{var-lib-cvmfs,var-run-cvmfs,home}
export SINGULARITY_BIND="/tmp/$USER/var-run-cvmfs:/var/run/cvmfs,/tmp/$USER/var-lib-cvmfs:/var/lib/cvmfs"
export SINGULARITY_HOME="/tmp/$USER/home:/home/$USER"
export EESSI_PILOT="container:cvmfs2 pilot.eessi-hpc.org /cvmfs/pilot.eessi-hpc.org"
singularity shell --fusemount "$EESSI_PILOT" docker://ghcr.io/bedroge/client-pilot:centos7
```

**Comment by bedroge** [comment]
It's now using a cache for the Docker layers, so it doesn't have to do the 3h rebuild every time. The last check (where nothing was changed) completed in ~1 minute instead of ~3h :-)

File: containers/build-or-download-cvmfs-debs.sh [hunk 1]
```diff
@@ -5,7 +5,10 @@ apt-get update
 apt-get install -y wget
 if [ "$arch" = "ppc64el" ] || [ "$arch" = "arm64" ]
```
Comment (amadio) [inline containers/build-or-download-cvmfs-debs.sh:6]: There's a typo in `ppc64le` above.

Comment (bedroge) [reply containers/build-or-download-cvmfs-debs.sh:6]: That's actually a Debian thing, really annoying:
https://wiki.debian.org/ppc64el

I was using $(uname -m) everywhere, which returns `aarch64`, `ppc64le`, and `x86_64`. But for Debian packages they, apparently, use `amd64`, `arm64`, and `ppc64el`. :-(

Comment (bedroge) [reply containers/build-or-download-cvmfs-debs.sh:6]:
```
Singularity> dpkg --print-architecture
ppc64el
```

**Comment by boegel** [comment]
@bedroge Merge conflicts need to be dealt with here...

**Comment by bedroge** [comment]
That should be easy (they can be removed), but the step for building the fuse-overlayfs images requires some more work.

**Comment by bedroge** [comment]
@boegel I think this should be okay now. The client container does still include `fuse-overlayfs` too; it's just one (small) executable and could maybe be useful? Or shall I remove it?

**Review by boegel — APPROVED** [review]

**Pull request merged**